test/recurse.js

```diff
@@ -7,6 +11,10 @@ assert.deepEqual(requireDir('./recurse'), {
 });
 
 // then test with recursing:
+var dir = path.join(__dirname, 'recurse', 'node_modules');
+mkdirp.sync(dir);
+fs.writeFileSync(path.join(dir, 'a.js'), 'module.exports = "ignore";');
```

Comment:
Instead of creating this directory and file programmatically on every test run, let's just check them in! =)

How about `test/recurse/node_modules/fake.js`?